enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,455 @@
+  }
+}
+```
+
+If the compliance event is recorded correctly, an HTTP 201 status code is returned.
+
+##### Viewing Compliance Events
+
+Another HTTP endpoint would be added at `/api/v1/compliance-events` that would accept a `GET` request. This would be
+responsible for exposing the compliance events and performing the correct authorization. The authorization would require
+`get` permissions on the cluster's associated `ManagedCluster` object to be able to list the compliance events for the
+cluster. For performance reasons, this is preferred to performing a `SubjectAccessReview` on every cluster namespace for
+a user per request. We can utilize the
+[github.com/stolostron/rbac-api-utils](https://github.com/stolostron/rbac-api-utils/blob/159deac7d398c3470ae730c3fbb8ffbbcca1ac5a/pkg/rbac/rbac.go#L187)
+library to do this.
+
+The following optional query parameters would be accepted on this HTTP endpoint:
+
+- cluster - the name of the cluster to filter on. Multiple can be provided with `?cluster=local-cluster&cluster=managed`
+  syntax.
+- parent_policy - the name of the parent policy (i.e. a `Policy`) of the policy (e.g. a `ConfigurationPolicy`) that
+  generated the compliance event to filter on.
+- policy - the name of the policy (e.g. a `ConfigurationPolicy`) that generated the compliance event to filter on.
+- policy_kind - the kind of the policy (e.g. `ConfigurationPolicy`) that generated the compliance event to filter on.
+- policy_api_group - the `apiGroup` of the policy (e.g. `policy.open-cluster-management.io`) that generated the
+  compliance event to filter on.
+- before - an ISO-8601 timestamp to indicate only compliance events before this time should be shown.
+- after - an ISO-8601 timestamp to indicate only compliance events after this time should be shown.
+- page - the page number in the query. This defaults to 1.
+- per_page - the number of compliance events returned per page. This defaults to 20 and maxes out at 100.
+
+The output would look as follows and be sorted by timestamps in descending order. The content would be filtered based on
+the user's access as described above:
+
+```json
+{
+  "data": [
+    {
+      "cluster": {
+        "name": "cluster1"
+      },
+      "parent_policy": {
+        "name": "policies.etcd-encryption"
+      },
+      "policy": {
+        "apiGroup": "policy.open-cluster-management.io",
+        "kind": "ConfigurationPolicy",
+        "spec": {},
+        "specHash": {}
+      },
+      "event": {
+        "compliance": "NonCompliant",
+        "message": "violation - configmaps [app-data] not found in namespace default",
+        "timestamp": "2023-07-19T18:25:43.511Z",
+        "metadata": {}
+      }
+    },
+    {
+      "cluster": {
+        "name": "cluster2"
+      },
+      "parent_policy": {
+        "name": "policies.etcd-encryption"
+      },
+      "policy": {
+        "apiGroup": "policy.open-cluster-management.io",
+        "kind": "ConfigurationPolicy",
+        "spec": {},
```

Comment:
So the entire spec is in all of these output entries?

Reply:
Yes, I think that's the easiest way to consume the data. Do you have other suggestions?

Reply:
It feels a lot like the `managedFields` data for k8s resources.  It's nice to have it there and necessary to be able to get it, but most of the time you don't need to see it... Maybe an `includeSpec` option in the request would be nice to think about.